gateway-util-common/src/main/java/org/apache/knox/gateway/util/Urls.java

```diff
@@ -71,6 +71,16 @@ public static int dotOccurrences(String domain) {
     return domain.length() - domain.replace(".", "").length();
   }
 
+  /**
+   * Does the provided URL contain UserInfo
+   * @param url
+   * @return true if a URL contains userInfo else false
+   * @throws MalformedURLException
+   */
+  public static boolean containsUserInfo(String url) throws MalformedURLException {
+    return (new URL(url).getUserInfo() != null);
```

Comment:
Do we need to worry about encoded @ i.e `%40` ?

Reply:
The URL class takes care of any such requirements.
I also tried adding a unit test for it just for kicks and replacing it with %40 results in a MalformedURLException.
I think we are good.

Reply:
Cool!